battles/endpoints.py

```diff
@@ -1,11 +1,27 @@
-from rest_framework import generics
+from django.db.models import Q
+
+from rest_framework import generics, permissions
 
 from battles.models import Battle
 from battles.permissions import IsTrainerInBattle
-from battles.serializers import BattleReadSerializer
+from battles.serializers import BattleReadSerializer, ListBattleSerializer
 
 
 class DetailBattleEndpoint(generics.RetrieveAPIView):
     queryset = Battle.objects.all()
     serializer_class = BattleReadSerializer
     permission_classes = [IsTrainerInBattle]
+
+
+class ListBattleEndpoint(generics.ListAPIView):
+    queryset = Battle.objects.all()
```

Comment:
I think you don't need this since you are already adding the `get_queryset`

Reply:
Done.